File: Java/FindTheDifferences.java
import java.util.HashMap;

public class FindTheDifferences {
    static char findTheDifference(String s, String t) {
        HashMap<Character, Integer> count = new HashMap<>();
        for (char c : s.toCharArray()) {
            count.put(c, count.getOrDefault(c, 0) + 1);
        }
        for (char c : t.toCharArray()) {
            if (!count.containsKey(c) || count.get(c) == 0) {
                return c;
            }
            count.put(c, count.get(c) - 1);
        }
        
        return ' ';
    }
    public static void main(String[] args) {
        String s="abcd",t="abcde";
        System.out.println(findTheDifference(s, t)+" is the letter added");
    }
}


// Input: s = "abcd", t = "abcde"
// Output: "e"
// Input: s = "", t = "y"
// Output: "y"

File: LeetCode/Python/FindTheDifferences.py
from collections import Counter
def findTheDifference(s: str, t: str) -> str:
    
        count_s=Counter(s)
        count_t=Counter(t)
        for char in count_t:
            if count_t[char] != count_s.get(char, 0):
                return char
            
s='abcd'
t='abcde'
print(findTheDifference(s,t)," is the addesd letter")

# Input: s = "abcd", t = "abcde"
# Output: "e"
# Input: s = "", t = "y"
# Output: "y"